src/uct/ib/mlx5/ib_mlx5.inl

```diff
@@ -257,15 +274,8 @@ unsigned uct_ib_mlx5_set_data_seg_iov(uct_ib_mlx5_txwq_t *txwq,
         }
         ucs_assert(iov[iov_it].memh != UCT_INVALID_MEM_HANDLE);
 
-        /* check consistency of the circular buffer
-         * assume dptr_it 16 bytes aligned and can't go behind txwq->end
-         */
-        ucs_assert(0 == (((uintptr_t)dptr_it) % sizeof(*dptr)));
-        if (ucs_unlikely(dptr_it == txwq->qend)) {
-            dptr_it = txwq->qstart;
-        }
-        ucs_assert((void*)dptr_it >= txwq->qstart);
-        ucs_assert((void*)(((struct mlx5_wqe_data_seg *)dptr_it) + 1) <= txwq->qend);
+        /* assume dptr_it 16 bytes aligned and can't go behind txwq->end */
+        dptr_it = uct_ib_mlx5_txwq_wrap_exact(txwq, dptr_it);
```

Comment:
Yes, if `dptr_it` is not behind the `qend` already (by previous pointer calculations).